packages/core/core/src/Transformation.js

```diff
@@ -492,36 +511,52 @@ export default class Transformation {
     filePath: FilePath,
     isSource: boolean,
     pipeline: ?string,
+    inputEnv: InternalEnvironment,
   ): Promise<Pipeline> {
+    let transformers = await this.parcelConfig.getTransformers(
+      filePath,
+      pipeline,
+      this.request.isURL,
+    );
+
+    // Allow transformers in the pipeline to override the requested environment.
+    // For example, some transformers may not be affected by certain options
+    // (e.g. library mode, scope hoisting, or output formats).
+    let env = transformers.reduce((env, {plugin}) => {
```

Comment:
Does this mean that e.g. with Less asset, the LessTransformer would see `asset.scopeHoist === true` because it's only cleared in the following `*.css` pipeline (or something like that for 3 chained pipelines)? I guess there's nothing we can do about this though...

Reply:
2. In my mind, it should rather be "if every transformer doesn't care about `scopeHoist`, then it can be dropped". Instead of `if any transformer.....`
3. In this case
```json
{
	"*.a": ["parcel-transformer-a-to-b", "parcel-transformer-unused"],
	"*.b": ["parcel-transformer-b"]
}
```
`parcel-transformer-unused` never runs because the asset type changes before it's reached. But `getEnvironment` is still applied, which might be unexpected as well.